packages/backend/src/dbt/dbtCliClient.ts

```diff
@@ -219,6 +226,13 @@ export class DbtCliClient implements DbtClient {
         );
     }
 
+    private static validateSelector(selector: string): boolean {
```

Comment:
> While execa does handle argument escaping to prevent shell injection (it uses child_process.spawn directly rather than going through a shell by default), it's still good practice to validate the selector input since it's being passed to dbt commands.